File: cpp/bank.cpp
#include <iostream>
#include <string>
using namespace std;

class BankAccount{

    private:
    string holder;
    long no;
    float balance;
    float dep;
    float with;

    public:

    BankAccount(string h, long n, float b){
        holder=h;
        no = n;
        balance = b;
    }

    void display(){
        
        cout<<"Account holder name: " << holder<<endl;
        cout<<"Account Number: " << no<<endl;
        cout<<"Current Balance: " << balance<<endl;
    }

    void deposit(float dep){
        balance += dep;
        cout<<"Deposited " << dep << " into the account" <<endl; 
    }

    void withdraw(float with){
            
        if(balance-with < 0){
            cout<<"Insufficient balance or invalid withdrawal amount"<<endl;
        }

        else{
            balance -= with;
            cout<<"Withdrawn "<<with<<" from the account"<<endl;
        }

    }
    
};

int main(){
    string holder;
    float balance, with[2], dep;
    long no;

    cout<<"Enter account holder name: ";
    cin>>holder;
    cout<<"Enter account number: ";
    cin>>no;
    cout<<"Enter account balance: ";
    cin>>balance;
    cout<<"Enter deposit amount: ";
    cin>>dep;
    cout<<"Enter withdrawl amount 1: ";
    cin>>with[0];
    cout<<"Enter withdrawal amount 2: ";
    cin>>with[1];

    BankAccount account(holder, no, balance);
    
    account.display();
    account.deposit(dep);
    account.withdraw(with[0]);
    account.withdraw(with[1]);
    account.display();
  

    return 0;


}

File: Python/bank.py
class bankaccount:
    def __init__(self, account_number, account_name, balance):
        self.account_number = account_number
        self.account_name = account_name
        self.balance = balance


    def deposit(self,cash):
        balance = cash + balance
        return balance

    def withdraw(self, cash):
        if balance < cash:
            return "Insufficient balance"
        else:
            balance = balance - cash
            return balance
    

print("Welcome to bank!")
a=int(input("Enter  your account number: "))
b=str(input("Enter your account name:"))
c=float(input("Enter current balance: "))

a1 = bankaccount(a,b,c)

choice = 0

while(choice != 4):

    choice = int(input("\nWhat operation to perform?\n1.View account details\n2.Deposit money\n3.Withdraw money\n4.Exit menu\n"))


    if choice == 1:
        print("Account Number: ", a1.account_number)
        print("Account Name: ", a1.account_name)
        print("Account Balance: ", a1.balance)
    elif choice == 2:
        dep=float(input("Enter cash to deposit: "))
        a1.deposit(dep)
    elif choice == 3:
        withd=float(input("Enter cash to withdraw: "))
        a1.withdraw(withd)
    elif choice != 4:
        print("Invalid choice. Try again.")



